azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -1002,12 +1003,30 @@ private void pollExecution() {
               .selectAndUpdateExecution(this.executorId, FlowRunnerManager.this.active);
           if (execId != -1) {
             FlowRunnerManager.logger.info("Submitting flow " + execId);
-            submitFlow(execId);
-            FlowRunnerManager.this.commonMetrics.markDispatchSuccess();
+            try {
+              submitFlow(execId);
+              FlowRunnerManager.this.commonMetrics.markDispatchSuccess();
+              this.numRetries = 0;
+            } catch (final ExecutorManagerException e) {
+              // If the flow fails to be submitted, then unset its executor id in DB so that other
+              // executors can pick up this flow and submit again.
+              FlowRunnerManager.this.executorLoader.unsetExecutorIdForExecution(execId);
```

Comment:
We are calling unsetExecutorIdForExecution() method from FlowRunner.createFlowRunner() as well. Consider removing the call from there to avoid unsetting executor id twice.

Reply:
Good point. Unfortunately, `FlowRunner.createFlowRunner()` is also called from `ExecutorManager` as well, removing it would mess up with the old dispatching logic.